### ISSUE TITLE
API Errors not localized

### ISSUE THREAD

**Issue by nmago** [boxyhq/saas-starter-kit#1285]
Hello, guys!

Thank you for putting in such effort into the template codebase; the thing seems amazing.

I have a localization-related question. Upon my initial examination of the codebase, I noticed that API errors aren't localized and are always returned in plain English.

For example, in the file **pages/auth/forgot-password.tsx**:

```javascript
if (!response.ok) {
  toast.error(json.error.message);
  return;
}

toast.success(t('password-reset-link-sent'));
```

While the success message is being translated, the API error is being passed as is.

Is this intentional or left for the future? Maybe I'm missing something?

**Comment by Musyonchez** [comment]
Can I get assigned to this.

**Comment by deepakprabhakara** [comment]
Thanks @Musyonchez, assigned the issue to you.

**Comment by SharmaPrakhar25** [comment]
Greetings @deepakprabhakara, Can I pick this up ? I can see that @Musyonchez has an open PR with pending work and is not working on it anymore.

**Comment by hkrhemendra** [comment]
Hi @deepakprabhakara , I looking for my first open source contribution. 
Can I pick this issue?

**Comment by deepakprabhakara** [comment]
Apologies for the delay in getting to this, @SharmaPrakhar25 please pick this up. Many thanks.

**Comment by siddharth122002** [comment]
Hi @deepakprabhakara , in this issue we only need to call t function for errors, right?

**Comment by Tanveer-rajpurohit** [comment]
"Hey @nmago  @deepakprabhakara , I've made the changes for API error localization. Let me know if it looks good or if anything needs to chnage. Thanks!"
